### ISSUE TITLE
Long option with an empty value is not parsed correctly

### ISSUE THREAD

**Issue by brunonova** [jopt-simple/jopt-simple#116]
When I provide an option in the command-line with an empty value in the form `--arg=`, jopt-simple is not parsing it correctly.

Here's an example showing this problem:

```java
OptionParser parser = new OptionParser();
parser.accepts("arg1").withRequiredArg();
parser.accepts("arg2").withRequiredArg();
OptionSet options = parser.parse(new String[] {"--arg1=", "--arg2=2"});

System.out.println("arg1: " + options.valueOf("arg1"));
System.out.println("arg2: " + options.valueOf("arg2"));
```

I was expecting the output of this code to be:

```
arg1: 
arg2: 2
```

Instead, the output was:

```
arg1: --arg2=2
arg2: null
```

Unless I'm missing something, this doesn't seem right to me.
It seems the empty value of `arg1` is being ignored, and `--arg1=` is being treated as `--arg1`.

**Comment by pholser** [comment]
@brunonova Thanks for your interest in JOpt Simple!

I wonder if we ought to consider your case a parsing error. If you're going to use the `=` form of specifying arguments to options, maybe we should require a value be there.

If you want to allow the zero-length string as an argument of an option, you could use:

    `--arg1 '' --arg2=2`

What do you think about enforcing a value immediately after `=`?

**Comment by lbergelson** [comment]
I would have assumed that `--arg1=` is a parsing error.

**Comment by brunonova** [comment]
I'm not sure, but I think I already used arguments of type `--arg1=` in other commands (in the command-line itself).

For example, something like `git log --decorate= --max-count=2` seems to work fine.
And `getopt -l arg1:,arg2: -- prog --arg1= --arg2=b` prints ` --arg1 '' --arg2 'b' --`.

So maybe `--arg1=` should be considered valid syntax to specify an empty value?

**Comment by jlmuir** [comment]
I ran into this same issue.  I would consider `--arg1=` not to be a parsing error but rather to indicate an empty value.  That's how the [glibc getopt-long example](https://www.gnu.org/software/libc/manual/html_node/Getopt-Long-Option-Example.html) behaves (`d` and `a` are the short forms of `delete` and `add`, respectively):

```
% ./main --delete= --add
option -d with value `'
option -a
```

I submitted pull request #122 as a fix for this, but I wouldn't mind at all, @pholser, if you'd rather fix it in a different way.

**Comment by pholser** [comment]
@brunonova @jlmuir This should be in 5.0.next, and master.

**Comment by brunonova** [comment]
That's great! Thanks @pholser and @jlmuir!

**Comment by twoqubed** [comment]
@brunonova This looks to be a duplicate of https://github.com/jopt-simple/jopt-simple/pull/122, which has the fix available in `5.0.4`.
